File: tests/testdata.cpp
#include "testdata.h"
#include <swlib/sw.h>
#include <array>
#include <string_view>
#include <chrono>

namespace testdata {

//
// All entries are invalid
//
std::string_view all_invalid{R"(
] BackwardsBrackets [
1765721157118

NoBrackets
1765721157118

[ MissingRightBracket
1765721157118

MissingLeftBracket]
1765721157118

[MissingTimestamp]


)"};


//
// All entries are valid
//
std::string_view all_valid{R"([whatever]
1765721157118

[name with spaces]
1765721157118


[Name~with " special' ;chars!@#$%^&]
1765721157118

)"};

std::array<timer_entry,3> all_valid_entries{{
    {.timer_name="whatever", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))},
    {.timer_name="name with spaces", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))},
    {.timer_name="Name~with \" special' ;chars!@#$%^&", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))}
}};

std::string_view all_valid_reencoded{R"([whatever]
1765721157118

[name with spaces]
1765721157118

[Name~with " special' ;chars!@#$%^&]
1765721157118

)"};


//
// Mixed valid and invalid entries
//
std::string_view mixed_valid_invalid{R"([whatever]
1765721157118

[]
1765721157118

[name with spaces]
1765721157118

MissingLeftBracket]
1765721157118

1765721157118

[Name~with " special' ;chars!@#$%^&]
1765721157118

[   name with leading&trailing spaces  ]
1765721157118

)"};

std::array<timer_entry,4> mixed_valid_invalid_entries{{
    {.timer_name="whatever", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))},
    {.timer_name="name with spaces", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))},
    {.timer_name="Name~with \" special' ;chars!@#$%^&", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))},
    {.timer_name="   name with leading&trailing spaces  ", .start_time=std::chrono::system_clock::time_point(std::chrono::milliseconds(1765721157118))}
}};

std::string_view mixed_valid_invalid_reencoded{R"([whatever]
1765721157118

[name with spaces]
1765721157118

[Name~with " special' ;chars!@#$%^&]
1765721157118

[   name with leading&trailing spaces  ]
1765721157118

)"};

} // namespace testdata

File: csharp/tests/Sw.tests/TestData.cs
using System.Globalization;

internal static class TestData
{
    // All these example files use the same timestamp
    public static DateTimeOffset GetTimestamp()
    {
        DateTimeOffset.TryParseExact("2025-12-17T14:39:06.7866990+00:00",
            "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind,
            out DateTimeOffset startedUtc);
        return startedUtc;
    }
    //
    // All entries are invalid
    //
    public static readonly string AllInvalid = """
        ] BackwardsBrackets [
        2025-12-17T14:39:06.7866990+00:00

        NoBrackets
        2025-12-17T14:39:06.7866990+00:00

        [ MissingRightBracket
        2025-12-17T14:39:06.7866990+00:00

        MissingLeftBracket]
        2025-12-17T14:39:06.7866990+00:00

        [MissingTimestamp]


        """;


    //
    // All entries are valid
    //
    public static readonly string AllValid = """
        [whatever]
        2025-12-17T14:39:06.7866990+00:00

        [name with spaces]
        2025-12-17T14:39:06.7866990+00:00


        [Name~with " special' ;chars!@#$%^&]
        2025-12-17T14:39:06.7866990+00:00

        """;

    public static readonly List<Sw.TimerEntry> AllValidEntries = new List<Sw.TimerEntry> {
        new Sw.TimerEntry { TimerName = "whatever", StartTimeUtc = GetTimestamp() },
        new Sw.TimerEntry { TimerName = "name with spaces", StartTimeUtc = GetTimestamp() },
        new Sw.TimerEntry { TimerName = "Name~with \" special' ;chars!@#$%^&", StartTimeUtc = GetTimestamp() }
    };

    public static readonly string AllValidReencoded = """
        [whatever]
        2025-12-17T14:39:06.7866990+00:00

        [name with spaces]
        2025-12-17T14:39:06.7866990+00:00

        [Name~with " special' ;chars!@#$%^&]
        2025-12-17T14:39:06.7866990+00:00


        """;


    //
    // Mixed valid and invalid entries
    //
    public static readonly string MixedValidInvalid = """
        [whatever]
        2025-12-17T14:39:06.7866990+00:00

        []
        2025-12-17T14:39:06.7866990+00:00

        [name with spaces]
        2025-12-17T14:39:06.7866990+00:00

        MissingLeftBracket]
        2025-12-17T14:39:06.7866990+00:00

        2025-12-17T14:39:06.7866990+00:00

        [Name~with " special' ;chars!@#$%^&]
        2025-12-17T14:39:06.7866990+00:00

        [   name with leading&trailing spaces  ]
        2025-12-17T14:39:06.7866990+00:00

        """;

    public static readonly List<Sw.TimerEntry> MixedValidInvalidEntries = new List<Sw.TimerEntry> {
        new Sw.TimerEntry { TimerName = "whatever", StartTimeUtc = GetTimestamp() },
        new Sw.TimerEntry { TimerName = "name with spaces", StartTimeUtc = GetTimestamp() },
        new Sw.TimerEntry { TimerName = "Name~with \" special' ;chars!@#$%^&", StartTimeUtc = GetTimestamp() },
        new Sw.TimerEntry { TimerName = "   name with leading&trailing spaces  ", StartTimeUtc = GetTimestamp() }
    };

    public static readonly string MixedValidInvalidReencoded = """
        [whatever]
        2025-12-17T14:39:06.7866990+00:00

        [name with spaces]
        2025-12-17T14:39:06.7866990+00:00

        [Name~with " special' ;chars!@#$%^&]
        2025-12-17T14:39:06.7866990+00:00

        [   name with leading&trailing spaces  ]
        2025-12-17T14:39:06.7866990+00:00


        """;
}


